LibTrixi.jl/src/api_jl.jl

```diff
@@ -49,3 +49,53 @@ function trixi_step_jl(simstate)
 
     return nothing
 end
+
+function trixi_ndims_jl(simstate)
+    mesh, _, _, _ = Trixi.mesh_equations_solver_cache(simstate.semi)
+    return ndims(mesh)
+end
+
+function trixi_nelements_jl(simstate)
+    _, _, solver, cache = Trixi.mesh_equations_solver_cache(simstate.semi)
+    return Trixi.nelements(solver, cache)
+end
+
+function trixi_nvariables_jl(simstate)
+    _, equations, _, api_jl = Trixi.mesh_equations_solver_cache(simstate.semi)
+    return Trixi.nvariables(equations)
+end
+
+function trixi_get_cell_averages_jl(data_, simstate)
+
+    mesh, equations, solver, cache = Trixi.mesh_equations_solver_cache(simstate.semi)
+    nelements = Trixi.nelements(solver, cache)
+    nvariables = Trixi.nvariables(equations)
+
+    # convert C to julia
+    data = unsafe_wrap(Array, data_, nelements*nvariables)
+
+    u_ode = simstate.integrator.u
+    u = Trixi.wrap_array(u_ode, mesh, equations, solver, cache)
+
+    for (index, element) in enumerate(Trixi.eachelement(solver, cache))
+
+        # temporary storage for mean value on current element for all variables
+        u_mean = zero(Trixi.get_node_vars(u, equations, solver, 1, 1, element))
+
+        # compute mean value using nodal dg values and quadrature
+        for j in Trixi.eachnode(solver), i in Trixi.eachnode(solver)
```

Comment:
This is not dimension agnostic. I think we should at least support 2D and 3D out of the box (but really also 1D).

You don't need to change it here, but maybe add a TODO note to the code and create an issue for it?

Reply:
Arg, that's true.